test/runtime/repository.test.js

```diff
@@ -0,0 +1,42 @@
+const should = require('should');
+const { Repository } = require('../../src/runtime/repository');
+const simpleELMInclude = require('./fixtures/SimpleLibraryWithInclude.json');
+const simpleELMDependency = require('./fixtures/SimpleDependentLibrary.json');
+
+const simpleIncludeIdentifier = simpleELMInclude.library.identifier;
+const simpleDependencyIdentifier = simpleELMDependency.library.identifier;
+
+describe('Repository', () => {
+  let simpleELMJson;
+  beforeEach(() => {
+    // Deep clone ELM JSON to modify properties
+    simpleELMJson = JSON.parse(JSON.stringify(simpleELMInclude));
+  });
+
+  it('should resolve with proper id and version', () => {
+    const repository = new Repository([simpleELMJson, simpleELMDependency]);
+    const mainLib = repository.resolve(simpleIncludeIdentifier.id, simpleIncludeIdentifier.version);
+
+    mainLib.includes[simpleDependencyIdentifier.id].should.not.be.undefined();
+  });
+
+  it('should fallback to id when version is missing', () => {
+    delete simpleELMJson.library.includes.def[0].version;
+
+    const repository = new Repository([simpleELMJson, simpleELMDependency]);
+    const mainLib = repository.resolve(simpleIncludeIdentifier.id, simpleIncludeIdentifier.version);
+
+    mainLib.includes[simpleDependencyIdentifier.id].should.not.be.undefined();
+  });
+
+  it('should fail when version is incorrect', () => {
+    simpleELMJson.library.includes.def[0].version = 'fake-version';
+
+    const repository = new Repository([simpleELMJson, simpleELMDependency]);
+
+    // When resolution can't happen, an error is thrown
+    should(() =>
+      repository.resolve(simpleIncludeIdentifier.id, simpleIncludeIdentifier.version)
+    ).throw();
```

Comment:
This confused me at first, as I couldn't figure out why it was throwing.  Thanks for walking me through it.  As discussed, it's throwing because PR #243 introduced a block of code that doesn't null-check the included library before trying to access its code systems.  So the behavior of throwing when constructing the main lib is actually unintentional.

I think that the overall approach for how included libraries' codesystems are handled needs to be revisited, but that is not your concern.  Still, would you mind fixing [this](https://github.com/mgramigna/cql-execution/blob/a559aea61afcedcd99496daf5092c5ea73a1a9fb/src/elm/library.js#L66) so it null checks the library before accessing its codesystems property?  That way we will no longer throw there.  And then update this test to check that the dependency lib is undefined from the main lib (since I guess that is what is expected)?

It also might be good to do direct `resolve` on the dependency lib and assert it is undefined (instead of only going through the main lib).

Reply:
Good call. I simplified the approach a bit to use one fixture and test resolve that way.

I also added another guard in the library constructor line that you linked.